### DISCUSSION TITLE
Role specific token

### DISCUSSION THREAD

**Discussion by utkarshgupta2419** [keycloak/keycloak#22483] [Q&A]
Hi, How to send a token request to keycloak for a particular role? Like instead of getting all the roles in token for a user, we can specify the role and get role specific token. Is there a way we can achieve it?

**Comment by christian-2** [comment]
Can you give a specific example of how you want the token (if it's an OIDC ID token) to look like? Or do you want an OAuth 2.0 access token that is constrained to particular roles (in this case discussion #22347 may hold a relevant example)?

**Reply by ug1924** [reply]
Hi. Right now the token consists of all roles assign to the user. So Basically I want to pass the role/roles in token request to keycloak and want to the token for that specific role.
